mathics/builtin/calculus.py

```diff
@@ -1039,6 +1040,68 @@ def apply(self, f, n, n0, evaluation, options={}):
         except:
             pass
 
+def find_root_secant(f, x0, x, opts, evaluation) -> (Number, bool):
```

Comment:
Is it not possbile under various conditions to hook into mpmath's routines for doing this? See https://mpmath.org/doc/current/calculus/optimization.html for example.

If we do so we would I suspect we would improve performance over, immediately provide more solvers, while reducing the chance of there being an error that we introduce.

Reply:
And in the improtant case where the end is Equal to 0 or some constant, can't we use Sympy's solve. See https://docs.sympy.org/latest/modules/solvers/solvers.html

Reply:
Indeed, the idea is to do the same as in NIntegrate. But I wanted to have a couple of "native" routines when the optimized routines are not available.

Reply:
Ok, that makes sense. But it doesn't look like we are trying to use any of the sympy or mathmp routines that core these. Is this correct?

Reply:
I haven't looked at the corresponding mpmath or sympy math routines. How do they compare with what is here code-wise?

Reply:
Before delegating the work on that routines, we should check if it is possible to pre-compile the target function, and its derivatives,  in a way that we can pass those compiled versions. Also, for the one-dimensional problem, the implemented algorithms are quite efficient, having a bottleneck in the function evaluation. On the other hand, FindRoot with vector variables is not implemented yet, so I think that the use of mpmath routines should be implemented at a later stage.